internal/scorer/config_test.go

```diff
@@ -0,0 +1,214 @@
+package scorer
+
+import (
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/ossf/criticality_score/internal/scorer/algorithm"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestInput_ToAlgorithmInput(t *testing.T) {
+	type fields struct {
+		Bounds       *algorithm.Bounds
+		Condition    *Condition
+		Field        string
+		Distribution string
+		Tags         []string
+		Weight       float64
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    *algorithm.Input
+		wantErr bool
+	}{
+		{
+			name: "unknown distribution error",
+			fields: fields{
+				Field: "test",
+			},
+			want:    &algorithm.Input{},
+			wantErr: true,
+		},
+		{
+			name: "distribution value set",
+			fields: fields{
+				Field:        "test",
+				Distribution: "linear",
+			},
+			want: &algorithm.Input{
+				Source:       algorithm.Value(algorithm.Field("test")),
+				Distribution: algorithm.LookupDistribution("linear"),
+			},
+			wantErr: false,
+		},
+		{
+			name: "Valid condition",
+			fields: fields{
+				Field:        "test",
+				Distribution: "linear",
+				Condition: &Condition{
+					FieldExists: "test",
+				},
+			},
+			want: &algorithm.Input{
+				Distribution: algorithm.LookupDistribution("linear"),
+			},
+			wantErr: false,
+		},
+		{
+			name: "Invalid condition",
+			fields: fields{
+				Field:        "test",
+				Distribution: "linear",
+				Condition: &Condition{
+					FieldExists: "test",
+					Not: &Condition{
+						FieldExists: "test",
+					},
+				},
+			},
+			want:    &algorithm.Input{},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
```

Comment:
nit: I usually use `test` as the value name in other tests, but I'm okay with `tt` as well.

`tt` is quite similar to `t` too.

Reply:
Changed `tt` to `test`